src/components/TreeFilter/VirtualizedTreeView.tsx

```diff
@@ -282,6 +292,27 @@ export class VirtualizedTreeView extends React.PureComponent<IVirtualizedTreeVie
         }
     }
 
+    private renderAsyncLoadingNode(loadingTreeNodeKey) { 
+        const style = {
+            height: this.props.rowHeight,
+            marginLeft: this.props.isSingleSelect ? 6 : 18
+        };
+            return (
+                <div
+                    key={loadingTreeNodeKey + '_Loading'}
+                    className="item-container loading-container"
+                    style={style}
+                >
+                    <Spinner className="tree-view-async-loading-spinner"

```

Comment:
The Spinner component has his own prop label so you can use it to write Loading...